chat-sdk-ui/src/main/java/co/chatsdk/ui/threads/ThreadDetailsActivity.java

```diff
@@ -86,19 +89,23 @@ protected void initViews() {
             actionBar.setHomeButtonEnabled(true);
         }
 
-        final View actionBarView = getLayoutInflater().inflate(R.layout.chat_sdk_activity_thread_details, null);
+        threadImageView = findViewById(R.id.chat_sdk_thread_image_view);
 
-        // Allow the thread name to be modified by a long click
-        actionBarView.setOnLongClickListener(v -> {
-            // TODO: Implement this
-            return true;
-        });
+        updateMetaData();
 
-        threadImageView = findViewById(R.id.chat_sdk_thread_image_view);
+        disposableList.add(ChatSDK.events().sourceOnMain()
+                .filter(NetworkEvent.filterType(EventType.ThreadMetaUpdated))
+                .subscribe(networkEvent -> updateMetaData()));
     }
 
-    protected void loadData () {
+    protected void updateMetaData() {
+        // TODO: permanently move thread name into meta data
+        ThreadMetaValue nameMetaValue = thread.metaValueForKey("name");
```

Comment:
Can you replace this hard coded "name" with a parameter like `Keys.name_of_parameter`

Reply:
I still see this `"name"` parameter - [here](https://github.com/chat-sdk/chat-sdk-android/pull/493/files).